web/packages/teleport/src/services/api/api.ts

```diff
@@ -92,7 +110,7 @@ const api = {
     const isAdminActionMfaError = isAdminActionRequiresMfaError(
       parseError(json)
     );
-    const isMfaHeaderPresent = params.headers?.[MFA_HEADER];
+    const isMfaHeaderPresent = customOptions.headers?.[MFA_HEADER];
     const shouldRetry = isAdminActionMfaError && !isMfaHeaderPresent;
```

Comment:
I believe that `isMfaHeaderPresent` was there to detect a situation when `fetchJson` is called again with the MFA header added. Since `fetchJsonWithMfaAuthnRetry` no longer inserts the header itself, we have to check for the existence of `webauthnAssertResp` instead.

```suggestion
    const shouldRetry = isAdminActionMfaError && !webauthnAssertResp;
```